src/main/java/seedu/address/model/deliverable/ModelDeliverableManager.java

```diff
@@ -99,23 +100,33 @@ public boolean hasDeliverable(Deliverable deliverable) {
     @Override
     public void deleteDeliverable(Deliverable target) {
         deliverableBook.removeDeliverable(target);
+        if (deliverableInView.isSameDeliverable(target)) {
+            deliverableInView = null;
+        }
     }
 
     @Override
     public void addDeliverable(Deliverable deliverable) {
         deliverableBook.addDeliverable(deliverable);
+        setDeliverableInView(deliverable);
         updateFilteredDeliverableList(PREDICATE_SHOW_ALL_DELIVERABLES);
     }
 
     @Override
     public void setDeliverable(Deliverable target, Deliverable editedDeliverable) {
         requireAllNonNull(target, editedDeliverable);
         deliverableBook.setDeliverable(target, editedDeliverable);
+        setDeliverableInView(editedDeliverable);
     }
 
     @Override
-    public void completeDeliverable(Deliverable target) {
```

Comment:
Perhaps we can include setDeliverableInView(target) for marking a deliverable done as well.

Reply:
Marking deliverable as done is implemented through setDeliverable which already includes setDeliverableInView

Reply:
Marking deliverable as done is implemented through setDeliverable which already includes setDeliverableInView